module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -1742,67 +1742,27 @@ protected function getItemStatusCodes($item)
-                $statusType = $parts[0];
-                $status = $parts[1] ?? '';
-
-                if ('Item' === $statusType || 'ItemType' === $statusType) {
-                    switch ($status) {
-                    case 'CheckedOut':
-                        $overdue = false;
-                        if (!empty($reason['due_date'])) {
-                            $duedate = $this->dateConverter->convert(
-                                'Y-m-d',
-                                'U',
-                                $reason['due_date']
-                            );
-                            $overdue = $duedate < time();
-                        }
-                        $statuses[] = $overdue ? 'Overdue' : 'Charged';
-                        break;
-                    case 'Lost':
-                        $statuses[] = 'Lost--Library Applied';
-                        break;
-                    case 'NotForLoan':
-                    case 'NotForLoanForcing':
-                        // NotForLoan is special: status has a library-specific
-                        // status number. Allow mapping of different status numbers
-                        // separately (e.g. Item::NotForLoan with status number 4
-                        // is mapped with key Item::NotForLoan4):
-                        $statusKey = $key . ($reason['status'] ?? '-');
-                        // Replace ':' in status key if used as status since ':' is
-                        // the namespace separator in translatable strings:
-                        $statuses[] = $this->itemStatusMappings[$statusKey]
-                            ?? $reason['code'] ?? str_replace(':', '_', $statusKey);
-                        break;
-                    case 'Transfer':
-                        $onHold = false;
-                        if (!empty($item['availability']['notes'])) {
-                            foreach (array_keys($item['availability']['notes'])
-                                as $noteKey
-                            ) {
-                                if ('Item::Held' === $noteKey) {
-                                    $onHold = true;
-                                    break;
-                                }
-                            }
+
+                // Check for a mapping method for the unavailability reason:
+                $methodName = 'GetStatusCode'
+                    . preg_replace('/[^a-zA-Z0-9_]/', '', $code);
+                if (method_exists($this, $methodName)) {
+                    $statuses[]
+                        = call_user_func([$this, $methodName], $code, $data, $item);
+                } else {
+                    if (!empty($data['code'])) {
+                        $statuses[] = $data['code'];
+                    } else {
+                        $parts = explode('::', $code, 2);
+                        if (isset($parts[1])) {
+                            $statuses[] = $parts[1];
```

Comment:
It looks like this is a change from the old logic, which would have done: `$statuses[] = $parts[1] ?? ''`. It now looks like empty status codes will be ignored rather than set as blank strings. This may well be more correct than before; I just wanted to note the change in behavior in case it was unintentional.

Reply:
True. This should never happen, so the change should have no effect, but in case we can't determine a proper status it's better to fall back to the 'Not Available' status below, I think.

Reply:
Agreed, that makes more sense!